webapp/views.py

```diff
@@ -453,21 +454,65 @@ def advantage_view():
 
             raise http_error
 
+        listings_response = advantage.get_marketplace_product_listings(
+            "canonical-ua"
+        )
+
+        product_listings = listings_response.get("productListings")
+
+        monthly_products = {}
+        yearly_products = {}
+        listings_to_products = {}
+
+        for listing in product_listings:
+            listings_to_products[listing["id"]] = listing["productID"]
```

Comment:
This is the other way around: you want to map from the product -> listingID + metric + quantity.